ibm/data_source_ibm_is_security_group_rules.go

```diff
@@ -0,0 +1,310 @@
+									},
+									"cidr_block": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The CIDR block. This property may add support for IPv6 CIDR blocks in the future. When processing a value in this property, verify that the CIDR block is in an expected format. If it is not, log an error. Optionally halt processing and surface the error, or bypass the resource on which the unexpected CIDR block format was encountered.",
+									},
+									"crn": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The security group's CRN.",
+									},
+									"deleted": &schema.Schema{
+										Type:        schema.TypeList,
+										Computed:    true,
+										Description: "If present, this property indicates the referenced resource has been deleted and providessome supplementary information.",
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"more_info": &schema.Schema{
+													Type:        schema.TypeString,
+													Computed:    true,
+													Description: "Link to documentation about deleted resources.",
+												},
+											},
+										},
+									},
+									"href": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The security group's canonical URL.",
+									},
+									"id": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The unique identifier for this security group.",
+									},
+									"name": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The user-defined name for this security group. Names must be unique within the VPC the security group resides in.",
+									},
+								},
+							},
+						},
+						"code": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Optional:    true,
+							Description: "The ICMP traffic code to allow.",
+						},
+						"type": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Optional:    true,
+							Description: "The ICMP traffic type to allow.",
+						},
+						"port_max": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Optional:    true,
+							Description: "The inclusive upper bound of TCP/UDP port range.",
+						},
+						"port_min": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Optional:    true,
+							Description: "The inclusive lower bound of TCP/UDP port range.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMIsSecurityGroupRulesRead(d *schema.ResourceData, meta interface{}) error {
+	secGrpId := d.Get("security_group_id").(string)
+	err := securityGroupRulesList(d, meta, secGrpId)
```

Comment:
this wrapper function is not needed. Can add the contents of securityGroupRulesList here directly

Reply:
done